pkg/gator/sync/test/test_test.go

```diff
@@ -0,0 +1,205 @@
+package test
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/cachemanager/parser"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/gator/fixtures"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/gator/reader"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestTest(t *testing.T) {
```

Comment:
I suggest separating the error cases into their own test, so you have:

```
func TestTest...

func TestTest_errors...
```

This will make the per-test struct simpler and make your test cases easier to follow as a reader.

Reply:
The first would only have success cases, failing if any error was returned.

Reply:
Neat idea, will do